freyja/tests/test_deconv.py

```diff
@@ -1,15 +1,51 @@
 import unittest
 from freyja.sample_deconv import *
 import pandas.testing as pdt
+import pandas.api.types as ptypes
+from numpy.random import negative_binomial
 
 class DeconvTests(unittest.TestCase):
     def test_buildLineageMap(self):
         mapDict = buildLineageMap()
         self.assertTrue('Alpha'==mapDict['B.1.1.7'] and 'Delta'==mapDict['AY.4'])
 
-    # def test_build_mix_and_depth(self):
-        
+    def test_build_mix_and_depth_plus_reindex(self):
+        df_barcodes = pd.read_csv('freyja/data/usher_barcodes.csv',index_col=0)
+        muts = list(df_barcodes.columns)
+        varFn='freyja/data/mixture.tsv'
+        depthFn='freyja/data/mixture.depth'
+        mix,depths = build_mix_and_depth_arrays(varFn,depthFn,muts)
+        #just making sure the files we've read in are ready for use
+        self.assertTrue(ptypes.is_float_dtype(mix))
+        self.assertTrue(ptypes.is_float_dtype(depths)) 
+
+        df_barcodes,mix,depths = reindex_dfs(df_barcodes,mix,depths)
+        pdt.assert_index_equal(df_barcodes.columns,mix.index)
+        pdt.assert_index_equal(df_barcodes.columns,depths.index)
+
+        self.assertFalse('20C' in df_barcodes.columns)
+
+    def test_constellation_mapping(self):
+        mapDict = buildLineageMap()
+        vals = [0.1,0.5,0.31,0.01,0.02,0.01]
+        strains = ['B.1.617.2','Q.3','B.1.427','A.2.5','B.1.1','B.1.1.7']
+        locDict = map_to_constellation(strains,vals,mapDict)
+        locDict_ideal = {'Alpha':0.51,'Epsilon':0.31,'Delta':0.1,'Other':0.02,'Aaron':0.01}
+        self.assertTrue(locDict==list(locDict_ideal.items()))
```

Comment:
What about `self.assertEqual(locDict, locDict_ideal)`? Ohhhh, wait, I think this relates to the comment above about `localDict` following `sorted`. Makes sense. I would recommend either changing the return datatype of `map_to_constellation` so that it does return `dict`, or keeping it as is (changing the local variable name noted above), and doing `self.assertEqual(loc, list(loc_ideal.items())`.

Of those, returning `dict` is better as insertion order is not guaranteed to be stable for `dict` for most versions of python or across architectures. What that means is the call to `list(some_dict.items())` may return things in a different order depending on "stuff".